src/network/receipt_envelope.rs

```diff
@@ -1,14 +1,15 @@
 use core::fmt;
 
-use alloy::consensus::TxReceipt;
-use alloy::network::eip2718::{Decodable2718, Encodable2718};
+use alloy::consensus::{AnyReceiptEnvelope, TxReceipt, TxType};
+use alloy::network::eip2718::{Decodable2718, Eip2718Error, Encodable2718};
 use alloy::primitives::Log;
 use serde::{Deserialize, Serialize};
 
 #[derive(Clone, Debug, PartialEq, Eq, Serialize, Deserialize)]
 #[serde(untagged)]
 pub enum ReceiptEnvelope<T = Log> {
     Native(alloy::consensus::ReceiptEnvelope<T>),
+    Eip712(AnyReceiptEnvelope<T>),
```

Comment:
Could you leave a comment (not doc) stating that we use `AnyReceiptEnvelope` for now due to the fact that `alloy::consensus::ReceiptEnvelope` cannot be decoded because of different transaction type, but for now we don't need any custom functionality for EIP712 receipt?